rdrand_asm.S

```diff
@@ -374,8 +382,36 @@ aes_round_keys:
 	.space	11*16
 	.size	aes_round_keys, .-aes_round_keys
 
+/*
+ * This is necessary to inform a linker that this code has IBT (Indirect
+ * Branch Tracking) feature of the Intel CET (Control-flow Enforcement
+ * Technology) enabled.
+ * See: https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#index-fcf-protection
+ */
+#if defined __CET__ && (__CET__ & 1)
+	.section	.note.gnu.property,"a"
+	.align	8
+	.long	1f - 0f		/* name length */
```

Comment:
Unsure I understand the magic numbers here.  why not 1 rather than 1f - 0f?  Also what name is this length referring to?

This pattern repeats below, it also seems odd to me that things like the pr_datasz is a short value.

Reply:
Yes, these are magic numbers, I've copied this section from https://sourceware.org/annobin/annobin.html/Test-cf-protection.html and https://github.com/gcc-mirror/gcc/blob/master/gcc/config/i386/cet.h.

"not 1 rather than 1f - 0f" - this is not a number 1, this is a distance between labels 0f and 1f, i.e. a size of `.string "GNU"`. And this is a name length which is being referred to.

Same way, pr_datasz is `.long` and it is a distance between labels 3f and 2f.

It looks like the most important fields in this section is `.long   0xc0000002` (CET flags type I guess) and `.long   0x3` ((CET flags value). It is a value stating IBT and SS are supported. This section is shown as following by readelf, please note the "IBT" mark:

```
$ readelf -n rdrand_asm.o
Displaying notes found in: .note.gnu.property
  Owner                Data size 	Description
  GNU                  0x00000010	NT_GNU_PROPERTY_TYPE_0
      Properties: x86 feature: IBT, SHSTK
```

Reply:
ah, their labels, not constants, thats the bit I was missing. thank you!